libs/SalesforceSDK/src/com/salesforce/androidsdk/security/SalesforceKeyGenerator.java

```diff
@@ -110,6 +111,32 @@ public static String getEncryptionKey(String name) {
         return encryptionKey;
     }
 
+    /**
+     * Returns the legacy encryption key. This should be called only as a means to migrate to the new key.
+     *
+     * @param name Unique name associated with this legacy encryption key.
+     * @return Legacy encryption key.
+     * @deprecated Will be removed in Mobile SDK 9.0.
+     */
+    public static String getLegacyEncryptionKey(String name) {
```

Comment:
Why do we need the name for? Doesn't Android generate a single key named with the value of INTERNAL_ENTROPY?

Reply:
Yes - for internal use cases, but apps might have supplied some other value for `name`. If they need to upgrade, they'll need to retrieve the old key.